src/query/completions.go

```diff
@@ -135,23 +142,18 @@ func getAllCompletions(config *core.Configuration, currentPackage, prefix, repoR
 // If 'test' is true it will similarly complete only targets that are tests.
 // If 'hidden' is true then hidden targets (i.e. those with names beginning with an underscore)
 // will be included as well.
-func Completions(graph *core.BuildGraph, packageName, prefix string, pkgs []string, binary, test, hidden bool) {
-	count := printLabelsInPackage(graph, packageName, prefix, binary, test, hidden)
+func Completions(graph *core.BuildGraph, completions *CompletionPackages, binary, test, hidden bool) []string {
+	labels := labelsInPackage(graph, completions.PackageToParse, completions.NamePrefix, binary, test, hidden)
 	// If we're printing binary targets, we might not match any targets in the parsed. If we only matched one other
 	// package, we should try and match binary targets in there.
-	if binary && count == 0 && len(pkgs) == 1 {
-		printLabelsInPackage(graph, pkgs[0], prefix, binary, test, hidden)
-		// This isn't totally correct as we should really recurse the entire algo. This is probably good enough though.
-		fmt.Printf("//%s\n", pkgs[0])
-	} else {
-		for _, pkg := range pkgs {
-			fmt.Printf("//%s\n", pkg)
-		}
+	if binary && len(labels) == 0 && len(completions.Pkgs) == 1 {
+		return labelsInPackage(graph, completions.Pkgs[0], completions.NamePrefix, binary, test, hidden)
 	}
+	return labels
 }
 
-func printLabelsInPackage(graph *core.BuildGraph, packageName, prefix string, binary, test, hidden bool) int {
-	count := 0
+func labelsInPackage(graph *core.BuildGraph, packageName, prefix string, binary, test, hidden bool) []string {
+	ret := make([]string, 0)
```

Comment:
super nit: would probably write just `var ret string` if it's going to explicitly be empty initially